src/Commands/NotificationQueuer.php

```diff
@@ -0,0 +1,266 @@
+
+  /**
+   * The entity type manager interface.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  private $entityTypeManager;
+
+  /**
+   * NotificationQueuer constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager interface.
+   * @param \Drupal\media_mpx\AuthenticatedClientFactory $authenticated_client_factory
+   *   The factory to load authenticated mpx clients.
+   * @param \Lullabot\Mpx\DataService\DataServiceManager $data_service_manager
+   *   The manager to discover data service classes.
+   * @param \Drupal\Core\Queue\QueueFactory $queue_factory
+   *   The factory to load the mpx notification queue.
+   * @param \Drupal\Core\State\StateInterface $state
+   *   The state backend for notification IDs.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, AuthenticatedClientFactory $authenticated_client_factory, DataServiceManager $data_service_manager, QueueFactory $queue_factory, StateInterface $state) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->authenticatedClientFactory = $authenticated_client_factory;
+    $this->dataServiceManager = $data_service_manager;
+    $this->queueFactory = $queue_factory;
+    $this->state = $state;
+  }
+
+  /**
+   * Listen for mpx notifications for a given media type.
+   *
+   * @param string $media_type_id
+   *   The media type ID to import for.
+   *
+   * @usage media_mpx-listen mpx_video
+   *   Listen for notifications for the mpx_video media type.
+   *
+   * @command media_mpx:listen
+   * @aliases mpxl
+   */
+  public function listen($media_type_id) {
+    // First, we find the last notification ID.
+    $media_type = $this->loadMediaType($media_type_id);
+    $media_source = DataObjectImporter::loadMediaSource($media_type);
+    $notification_id = $this->getNotificationId($media_source);
+
+    // Next, we fetch notifications, removing duplicates (such as multiple saves
+    // of an mpx object in a row).
+    $notifications = $this->fetchNotifications($media_source, $notification_id);
+    $notifications = $this->filterDuplicateNotifications($notifications);
+
+    // Take the notifications and store them in the queue for processing later.
+    $this->queueNotifications($media_type, $notifications);
+
+    // Let the next listen call start from where we left off.
+    $notification_key = $media_source->getPluginId() . '_notification_id';
+    $this->state->set($notification_key, end($notifications)->getId());
+
+    $this->io()->progressFinish();
+  }
+
+  /**
+   * Return the current notification ID, or -1 if one is not set.
+   *
+   * @param \Drupal\media\MediaSourceInterface $media_source
+   *   The media source the notification ID is for.
+   *
+   * @return int
+   *   The notification ID.
+   */
+  protected function getNotificationId(MediaSourceInterface $media_source): int {
+    // @todo should this really be state?
```

Comment:
I think so. We don't want this to be exported to configuration, nor to exist in other sites than the current one, right?

Reply:
I think it's going to depend on if there are separate "stage" accounts in mpx or not. Either way, the worst case is we reprocess some data, and our notification system always uses the current state of the data, not whatever the (stale) notification might say.